asyncstdlib/asynctools.py

```diff
@@ -324,3 +325,55 @@ async def compute_something_else() -> float:
     return __func(
         *[await arg for arg in args], **{k: await arg for k, arg in kwargs.items()}
     )
+
+
+def make_awaitable(function: Callable[..., T]) -> Callable[..., Awaitable[T]]:
+    """
+    Wraps any Callable, which allows to use it as Awaitable object
+
+    :param function - can be any Callable
+
+    :raise TypeError if function argument is not Callable
+
+    Example:
+
+    .. code-block:: python3
+
+        import asyncstdlib as a
+
+        def test1_sync(x, y):
+            ...
+
+        async def test1_async(x):
+            ...
+
+        @a.make_awaitable
+        def test2_sync(x, y):
+            ...
+
+        @a.make_awaitable
+        async def test2_async(x):
+            ...
+
+        async def main():
+            await a.make_awaitable(test1_sync)(x=1, y=2)
+            await a.make_awaitable(test1_async)(x=8)
+            await test2_sync(x=3, y=2)
+            await test2_async(x=7)
+
+        if __name__ == "__main__":
+            asyncio.run(main())
+    """
+    if not isinstance(function, Callable):
+        raise TypeError("function argument should be Callable")
+
```

Comment:
Please add a fast path for native coroutines here. (``iscoroutinefunction`` comes from ``inspect``)
https://github.com/maxfischer2781/asyncstdlib/blob/b48fb6363b2110c72cfd012fea84b399c6b2aab9/asyncstdlib/_core.py#L91-L92
A primary use-case for this helper is to force some input function to be awaitable, no matter whether it was before. But often enough the input already is a proper, fast coroutine and wrapping it only makes things slower.